procinfo/src/macos.rs

```diff
@@ -178,6 +178,11 @@ impl LocalProcessInfo {
 
             let exe_path = consume_cstr(&mut ptr)?.into();
 
+            // For some reason, sysctl sometimes puts null bytes
+            // between the executable field and the argv field
+            let not_nul = ptr.iter().position(|&c| c != 0)?;
```

Comment:
perhaps this logic should be swept up into the bottom of `consume_cstr` above?

It would be great if you could encapsulate the example you shared in the description of the PR into a unit test for this; it would make the PR a bit bigger, but it feels worthwhile to have at least two tests that show this working for data that uses 1x NUL and 2x NUL separators.

I think factoring out a function like: `parse_exe_and_argv_sysctl(buf: &[u8]) -> Option<(PathBuf, Vec<String>)>` would make this more easily testable; we'd pass the buffer slice that we compute on line 167 to this function and return the result. Lines 169 through 191 would be the body of this new function.

Reply:
That sounds like a good change, I'll make that and get back to this.